tempodb/backend/azure/azure.go

```diff
@@ -373,42 +373,42 @@ func (rw *readerWriter) readRange(ctx context.Context, name string, offset int64
 	return nil
 }
 
-func (rw *readerWriter) readAll(ctx context.Context, name string) ([]byte, blob.ETag, error) {
-	blobURL := rw.hedgedContainerURL.NewBlockBlobURL(name)
+func (rw *readerWriter) readAll(ctx context.Context, name string) ([]byte, azcore.ETag, error) {
+	blobClient := rw.hedgedContainerClient.NewBlockBlobClient(name)
 
-	var props *blob.BlobGetPropertiesResponse
-	props, err := blobURL.GetProperties(ctx, blob.BlobAccessConditions{}, blob.ClientProvidedKeyOptions{})
+	props, err := blobClient.GetProperties(ctx, &blob.GetPropertiesOptions{})
 	if err != nil {
 		return nil, "", err
 	}
 
-	destBuffer := make([]byte, props.ContentLength())
+	destBuffer := make([]byte, *props.ContentLength)
```

Comment:
more pointer derefs

Reply:
Action taken: we now return an error if `props.ContentLength == nil`.